### PR TITLE
wasmer: Add cargo hashes for Python 3.6/3.7/3.9 support

### PR DESCRIPTION
**Pull request by esotericnonsense**
- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
And what about python 3.7 and 3.9?

**Comment by esotericnonsense** [comment]
latest commit is a slight code clean up to avoid the annoying if-else chain

**Comment by esotericnonsense** [comment]
also use `pythonOlder` rather than `isPy3{6,7,8,9}`

**Comment by jonringer** [comment]
maintaining this is going to be difficult, is there a way we could have a script ran to generate the related hashes?

**Comment by esotericnonsense** [comment]
@jonringer makes sense; are there any other examples in nixpkgs I may be able to look at for reference?

I had just manually run something like this (actually opened in 4 terminals and ran in parallel)

```
versions=( "36" "37" "38" "39" ); for x in "${versions[@]}"; do nix-build -A "python${x}Packages.wasmer"; done
```

and then just copy pasted the hashes in. It seems far easier to re-use nix's output than to independently figure out the cargo hash and paste it in.

**Comment by SuperSandro2000** [comment]
I wonder why the hash changes at all. Is it selecting dependencies based on python version?

> are there any other examples in nixpkgs I may be able to look at for reference?

There are many update.sh files but I did not find any that generated a vendorSha directly. They generate it with vgo2nix or yaml2nix.

> It seems far easier to re-use nix's output than to independently figure out the cargo hash and paste it in.

Maybe we can try to build it in the update script and just put the hash in the file?

File: pkgs/development/python-modules/wasmer/default.nix [hunk 1]
```diff
@@ -20,7 +20,12 @@ let
       sha256 = "0302lcfjlw7nz18nf86z6swhhpp1qnpwcsm2fj4avl22rsv0h78j";
     };
 
-    cargoSha256 = "0d83dniijjq8rc4fcwj6ja5x4hxh187afnqfd8c9fzb8nx909a0v";
+    cargoSha256 = {
```
Comment (FRidh) [inline pkgs/development/python-modules/wasmer/default.nix:23]: does cargo, depending on python version, fetch different dependencies?

Comment (jonringer) [reply pkgs/development/python-modules/wasmer/default.nix:23]: another possibility is that there's some reference to the python version, thus the vendor libs get a different hash

Comment (danieldk) [reply pkgs/development/python-modules/wasmer/default.nix:23]: The problem is that the Python version is used in the `name` attribute of the Rust derivation. `name` is used as part of the vendor directory and tarball. So, if the Python version is part of the name, the vendored tarball changes for every Python version.

Fixed in #110580.

Comment (SuperSandro2000) [reply pkgs/development/python-modules/wasmer/default.nix:23]: Going to close this in favor of daniels PR.

**Pull request closed without merging**